### DISCUSSION TITLE
How to set file upload location? or set custom uploader option?

### DISCUSSION THREAD

**Discussion by 4n1qz5skwv** [litestar-org/litestar#1463] [Q&A]
Currently, Uploaded files are stored in temporary location. In order to save it elsewhere, we have to rewrite or copy it. 

Is it possible to setup a custom handler?

Say, uploaded file will saved in my_dir/{uuid4}/filename.ext directly ?

**Comment by Goldziher** [comment]
Not at present. 

We could potentially do this, what's the use case?

**Reply by 4n1qz5skwv** [reply]
> Not at present.

I see. We are using shutil copy at the moment. Can we change the `rolled-to-disk threshold` to 0 to always save in disk to have a filepath?

https://docs.litestar.dev/2/reference/datastructures.html#litestar.datastructures.UploadFile.rolled_to_disk

> what's the use case?

For now, We are trying to implement this,
https://pqina.nl/filepond/docs/api/server/#process

...
2. server saves file to unique location tmp/12345/my-file.jpg
3. server returns unique location id 12345 in text/plain response
...

**Reply by Goldziher** [reply]
Yes, we can add this. Please open an issue on GitHub? PR would be welcome too.

**Reply by 4n1qz5skwv** [reply]
Done. https://github.com/litestar-org/litestar/issues/1465
